pkg/pillar/cmd/downloader/syncop.go

```diff
@@ -356,7 +356,7 @@ func handleSyncOpResponse(ctx *downloaderContext, config types.DownloaderConfig,
 	_, err := os.Stat(locFilename)
 	if err != nil {
 		// Nothing was downloaded? Delete progress files if any
-		doDelete(ctx, key, locFilename, status)
+		doDelete(ctx, key, locFilename+progressFileSuffix, status)
```

Comment:
then we have to check the sane name several lines above...
```go
 _, err := os.Stat(locFilename)
```

Reply:
And don't we need to delete with and without the suffix?

Reply:
At line 356 we check if the downloaded file (locFilename) exists, if it doesn't, then we need tor remove any remained progress files, which is done at line 359. However, the doDelete() was being called with locFilename, which at this point we know it doesn't exist, the right file name is locFilename+progressFileSuffix (progress files), that's what this commit fixes... my understanding is that everything else in this function is correct...

Reply:
But it seems that `doDelete` handles removal of progress files: https://github.com/lf-edge/eve/blob/master/pkg/pillar/cmd/downloader/downloader.go#L438-L439

Reply:
Indeed @milan-zededa , doDelete() already handles both: file + progress file... in this case it will fail for the locFilename (which doesn't exist) but it doesn't care, it calls also for progress file... This change in the PR is not needed, definitely. However, I believe that is hard to assume that for doDelete() at first glance, so I just changed the PR to improve the comments, which IMHO, are not clear, but I want reviewers to feel free to reject this PR if they think it's not needed.